tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py

```diff
@@ -501,10 +510,12 @@ def test_syscollector_second_scan(
 
         global FIRST_SCAN_TIME
 
+        max_result_window(host_manager)
```

Comment:
I can't help but notice max_result_window is called several times yet the variable host_manager **doesn't seem** to be updated in any way (assignment in this file, modification in some function/method where it is passed as a parameter). Is this intentional?
If the reason to do this is increase several times the result window I believe it could be replaced by a single bigger increment.

Reply:
Except for this one that is called before `get_vulnerabilites_from_states_by_agent`
**Not sure**, but if it's always called with these functions it could be either added to those functions or create a wrapper function to call both functions together in the correct order.

Reply:
This approach was intentional. While it isn't the cleanest solution, several limitations have led us to adopt it:

- Extending the result window requires the index to exist, which rules out fixtures that perform this operation at the start of the test since the index is frequently deleted during the initial scans.
- The tests are designed to be run independently, so making changes in only one test could result in unpredictable behavior that's difficult to debug.
 - Although it may be possible to improve this logic by refactoring the tests, these tests will be completely revamped for version 5.0.0. Therefore, I recommend not investing time in this for now.